stdlib/platform.pyi

```diff
@@ -14,20 +15,46 @@ def java_ver(
 def system_alias(system: str, release: str, version: str) -> tuple[str, str, str]: ...
 def architecture(executable: str = sys.executable, bits: str = "", linkage: str = "") -> tuple[str, str]: ...
 
-if sys.version_info >= (3, 9):
-    @type_check_only
-    class _uname_result_base(NamedTuple):
+if sys.version_info >= (3, 10):
+    # uname_result emulates a 6-field named tuple, but the processor field
+    # is lazily evaluated rather than being passed in to the constructor.
+    # At runtime, it inherits from a 5-field NamedTuple instead of being a
+    # NamedTuple itself, but that's too hacky to represent in the stubs.
+    class uname_result(NamedTuple):
         system: str
         node: str
         release: str
         version: str
         machine: str
-
-    class uname_result(_uname_result_base):
+        processor: str  # actually a property
+        def __init__(self, system: str, node: str, release: str, version: str, machine: str) -> None: ...  # type: ignore[misc]
+        def __new__(_cls, system: str, node: str, release: str, version: str, machine: str) -> Self: ...  # type: ignore[misc]
         @property
-        def processor(self) -> str: ...
+        def __match_args__(
+            self,
+        ) -> tuple[Literal["system"], Literal["node"], Literal["release"], Literal["version"], Literal["machine"]]: ...
+
+elif sys.version_info >= (3, 9):
+    # uname_result emulates a 6-field named tuple, but the processor field
+    # is lazily evaluated rather than being passed in to the constructor.
+    # At runtime, it inherits from a 5-field NamedTuple instead of being a
+    # NamedTuple itself, but that's too hacky to represent in the stubs.
+    class uname_result(NamedTuple):
+        system: str
+        node: str
+        release: str
+        version: str
+        machine: str
+        processor: str  # actually a property
+        def __init__(self, system: str, node: str, release: str, version: str, machine: str) -> None: ...  # type: ignore[misc]
+        def __new__(_cls, system: str, node: str, release: str, version: str, machine: str) -> Self: ...  # type: ignore[misc]
+        @property  # type: ignore[misc]
+        def _fields(
+            self,
+        ) -> tuple[Literal["system"], Literal["node"], Literal["release"], Literal["version"], Literal["machine"]]: ...
```

Comment:
If the difference between 3.9 and 3.10 is only on one property, I think this could be shortened to the following (which imo makes it clearer where it defers)
```
if sys.version_info >= (3, 9):
    # uname_result emulates a 6-field named tuple, but the processor field
    # is lazily evaluated rather than being passed in to the constructor.
    # At runtime, it inherits from a 5-field NamedTuple instead of being a
    # NamedTuple itself, but that's too hacky to represent in the stubs.
    class uname_result(NamedTuple):
        system: str
        node: str
        release: str
        version: str
        machine: str
        processor: str  # actually a property
        def __init__(self, system: str, node: str, release: str, version: str, machine: str) -> None: ...  # type: ignore[misc]
        def __new__(_cls, system: str, node: str, release: str, version: str, machine: str) -> Self: ...  # type: ignore[misc]
        if sys.version_info >= (3, 10):
            @property
            def __match_args__(
                self,
            ) -> tuple[Literal["system"], Literal["node"], Literal["release"], Literal["version"], Literal["machine"]]: ...
        else:
            @property  # type: ignore[misc]
            def _fields(
                self,
            ) -> tuple[Literal["system"], Literal["node"], Literal["release"], Literal["version"], Literal["machine"]]: ...
```
In fact, even the 3.8 branch could be merged in:
```py
class uname_result(NamedTuple):
    system: str
    node: str
    release: str
    version: str
    machine: str
    processor: str  # actually a property on 3.9+

    # uname_result emulates a 6-field named tuple, but the processor field
    # is lazily evaluated rather than being passed in to the constructor.
    # At runtime, it inherits from a 5-field NamedTuple instead of being a
    # NamedTuple itself, but that's too hacky to represent in the stubs.
    if sys.version_info >= (3, 9):
        def __init__(self, system: str, node: str, release: str, version: str, machine: str) -> None: ...  # type: ignore[misc]
        def __new__(_cls, system: str, node: str, release: str, version: str, machine: str) -> Self: ...  # type: ignore[misc]
        if sys.version_info >= (3, 10):
            @property
            def __match_args__(
                self,
            ) -> tuple[Literal["system"], Literal["node"], Literal["release"], Literal["version"], Literal["machine"]]: ...
        else:
            @property  # type: ignore[misc]
            def _fields(
                self,
            ) -> tuple[Literal["system"], Literal["node"], Literal["release"], Literal["version"], Literal["machine"]]: ...
```

Thoughts ?

Reply:
Unfortunately it won't work, because the conditionals don't work inside a NamedTuple. The `else:` line gets parsed incorrectly by mypy and generates `error: Invalid statement in NamedTuple definition; expected "field_name: field_type [= default]"  [misc]`

Reply:
Doesn't look like there's actually a ticket for that right now, so opened one: https://github.com/python/mypy/issues/18187